libs/wingc/src/type_check.rs

```diff
@@ -2113,18 +2278,24 @@ impl<'a> TypeChecker<'a> {
 						_ => {}
 					};
 
-          if !t.is_json_legal_value() {
+          if self.is_in_mut_json && !known_type.is_json_legal_value() {
             self.spanned_error(
               v,
               format!(
                 "Expected \"Json\" elements to be Json values (https://www.json.org/json-en.html), but got \"{}\" which is not a Json value",
-                t
+                known_type
               ),
             );
```

Comment:
style: use tabs, not spaces

(can we set up an automated linter for this?)

Reply:
Interesting, don't we already run rustfmt? My editor does as well, surprised this made it in, will look into it

Reply:
> Okay it turns out fmt stops working when the line is too long (max_width is set to 120). Unfortunately, it also doesn't provide any feedback about that

interesting! Yeah I was wondering why it seemed like `fmt` just stopped working sometimes... that explains a lot